src/envs/playroom.py

```diff
@@ -0,0 +1,845 @@
+from predicators.src.settings import CFG
+from predicators.src import utils
+
+
+class PlayroomEnv(BaseEnv):
+    """Boring room vs. playroom domain.
+    """
+    # Parameters that aren't important enough to need to clog up settings.py
+    x_lb = 0.0
+    y_lb = 0.0
+    x_ub = 140.0
+    y_ub = 30.0
+    table_height = 0.2
+    table_tol = 1.0
+    table_x_lb = 10.0
+    table_y_lb = 10.0
+    table_x_ub = 20.0
+    table_y_ub = 20.0
+    door_r = 5.0  # half of width
+    door_button_z = 1.0  # relative to ground
+    door_tol = 0.2
+    dial_r = 3.0
+    dial_tol = 1.0
+    dial_button_tol = 0.2
+    held_tol = 0.5
+    clear_tol = 0.5
+    open_fingers = 0.8
+    pick_tol = 0.4
+    assert pick_tol < CFG.playroom_block_size
+    lift_amt = 1.0
+
+    def __init__(self) -> None:
+        super().__init__()
+        # Types
+        self._block_type = Type(
+            "block", ["pose_x", "pose_y", "pose_z", "held", "clear"])
+        self._robot_type = Type("robot", ["pose_x", "pose_y", "rotation",
+                                          "fingers"])
+        self._door_type = Type("door", ["pose_x", "pose_y", "open"])
+        self._dial_type = Type("dial", ["pose_x", "pose_y", "level"])
+        # Predicates
+        self._On = Predicate(
+            "On", [self._block_type, self._block_type], self._On_holds)
+        self._OnTable = Predicate(
+            "OnTable", [self._block_type], self._OnTable_holds)
+        self._GripperOpen = Predicate(
+            "GripperOpen", [self._robot_type], self._GripperOpen_holds)
+        self._Holding = Predicate(
+            "Holding", [self._block_type], self._Holding_holds)
+        self._Clear = Predicate(
+            "Clear", [self._block_type], self._Clear_holds)
+        # Options
+        self._Pick = ParameterizedOption(
+            # variables: [robot, object to pick]
+            # params: [delta x, delta y, delta z]
+            "Pick", types=[self._robot_type, self._block_type],
+            params_space=Box(-1, 1, (3,)),
+            _policy=self._Pick_policy,
+            _initiable=self._Pick_initiable,
+            _terminal=self._Pick_terminal)
+        self._Stack = ParameterizedOption(
+            # variables: [robot, object on which to stack currently-held-object]
+            # params: [delta x, delta y, delta z]
+            "Stack", types=[self._robot_type, self._block_type],
+            params_space=Box(-1, 1, (3,)),
+            _policy=self._Stack_policy,
+            _initiable=self._Stack_initiable,
+            _terminal=self._Stack_terminal)
+        self._PutOnTable = ParameterizedOption(
+            # variables: [robot]
+            # params: [x, y] (normalized coordinates on the table surface)
+            "PutOnTable", types=[self._robot_type],
+            params_space=Box(0, 1, (2,)),
+            _policy=self._PutOnTable_policy,
+            _initiable=self._PutOnTable_initiable,
+            _terminal=self._PutOnTable_terminal)
```

Comment:
Aren't we missing some other predicates?

Reply:
yeah, will work on those (and operators and options) next!